sympy/solvers/solveset.py

```diff
@@ -1754,6 +1757,50 @@ def _is_logarithmic(f, symbol):
     return rv
 
 
+def _is_lambert(f, symbol):
+    r"""
+    Return True if the equation is of Lambert type, else False.
+    Equations containing `Pow`, `log` or `exp` terms are currently
+    treated as Lambert types.
+    """
+    return any(isinstance(arg, (Pow, exp, log)) for arg in _term_factors(f))
```

Comment:
This says "equations containing" but actually it is the code selects something more specific because it wouldn't accept e.g. `sin(exp(x))`. The description should be made precise.

Reply:
I've just taken the code from directly from https://github.com/sympy/sympy/pull/14972 , but in the next PR I'll change this code....I've done this just for the sake of simplicity...(to have min. diff.)

Reply:
> e.g. sin(exp(x))

I don't think so that this is of lambert type...

Reply:
No it's not of Lambert type. The docstring says "equations containing exp" though so it implies that `sin(exp(x))` would be of "lambert" type.

The docstring should give a clearer explanation of what this function is for that also matches what the code actually does.